### PR TITLE
fix(workflow): Throw if patches are used at Workflow top level

### LINKED ISSUE #362: [Bug] Patches should be applied to workflow before workflow code is required
We should support patches at the top level of user code as shown here:
```ts
let runActivity = false;
if (patched('p1')) {
  runActivity = true;
}

export async function myWorkflow() {
  if (runActivity) {
    await myActivity();
  }
}
```

### PR DESCRIPTION
**Pull request by bergundy**
Closes #362

### CODE REVIEW

File: packages/worker/src/worker.ts [hunk 1]
```diff
@@ -598,13 +598,29 @@ export class Worker {
                         isReplaying: activation.isReplaying,
                       };
 
+                      const patchJobs = activation.jobs.filter(
+                        (
+                          j
+                        ): j is NonNullableObject<
+                          Pick<coresdk.workflow_activation.IWFActivationJob, 'notifyHasPatch'>
+                        > => j.notifyHasPatch != null
```
Comment (Sushisource) [inline packages/worker/src/worker.ts:606]: Oh my god this is some nauseating formatting (I know it's auto-formatted, but, jesus)

File: packages/worker/src/workflow/isolated-vm.ts [hunk 2]
```diff
@@ -210,14 +195,13 @@ export class IsolatedVMWorkflow implements Workflow {
...
 
-    await context.evalClosure(
-      'lib.initRuntime($0, $1, $2, $3, $4)',
-      [info, interceptorModules, randomnessSeed, now, isolateExtension.derefInto()],
-      { arguments: { copy: true }, timeout: isolateExecutionTimeoutMs }
-    );
+    await context.evalClosure('lib.initRuntime($0, $1)', [options, isolateExtension.derefInto()], {
+      arguments: { copy: true },
+      timeout: isolateExecutionTimeoutMs,
+    });
```
Comment (Sushisource) [inline packages/worker/src/workflow/isolated-vm.ts:201]: That's nicer

File: packages/workflow/src/workflow.ts [hunk 3]
```diff
@@ -814,13 +814,13 @@ export function deprecatePatch(patchId: string): void {
...
-  if (state.info === undefined) {
-    throw new IllegalStateError('Workflow info must be set when calling patch functions');
-  }
   // Patch operation does not support interception at the moment, if it did,
   // this would be the place to start the interception chain
 
-  const { isReplaying } = state.info;
+  const { isReplaying } = workflowInfo();
+  if (state.workflow === undefined) {
+    throw new IllegalStateError('Patches cannot be used before Workflow starts');
```
Comment (Sushisource) [inline packages/workflow/src/workflow.ts:822]: Lift string to constant for assertion in test?

**Review by Sushisource — APPROVED** [review]

File: packages/worker/src/worker.ts (hunk 1 above)
Comment (bergundy) [reply packages/worker/src/worker.ts:606]: lol, I can declare the type somewhere else so the formatting is fixed

File: packages/workflow/src/workflow.ts (hunk 3 above)
Comment (bergundy) [reply packages/workflow/src/workflow.ts:822]: It's just 1 test, I can but I wouldn't bother

**Pull request merged**